src/types/mod.rs

```diff
@@ -72,32 +73,32 @@ macro_rules! pyobject_native_type_named (
                 unsafe { $crate::Py::from_borrowed_ptr(py, self.as_ptr()) }
             }
         }
-
-        impl<$($type_param,)*> From<&'_ $name> for $crate::Py<$name> {
-            #[inline]
-            fn from(other: &$name) -> Self {
-                use $crate::AsPyPointer;
-                use $crate::PyNativeType;
-                unsafe { $crate::Py::from_borrowed_ptr(other.py(), other.as_ptr()) }
-            }
-        }
     };
 );
 
 #[macro_export]
 macro_rules! pyobject_native_type_core {
-    ($name: ty, $layout: path, $typeobject: expr, $module: expr, $checkfunction: path $(,$type_param: ident)*) => {
+    ($name: ty, $layout: path, $typeobject: expr, $module: expr, $checkfunction: path) => {
         unsafe impl $crate::type_object::PyLayout<$name> for $layout {}
-        $crate::pyobject_native_type_named!($name $(,$type_param)*);
-        $crate::pyobject_native_type_convert!($name, $layout, $typeobject, $module, $checkfunction $(,$type_param)*);
-        $crate::pyobject_native_type_extract!($name $(,$type_param)*);
+        $crate::pyobject_native_type_named!($name);
+        $crate::pyobject_native_type_convert!($name, $layout, $typeobject, $module, $checkfunction);
+        $crate::pyobject_native_type_extract!($name);
 
-        impl<'a, $($type_param,)*> ::std::convert::From<&'a $name> for &'a $crate::PyAny {
+        impl<'a> ::std::convert::From<&'a $name> for &'a $crate::PyAny {
             fn from(ob: &'a $name) -> Self {
-                unsafe{&*(ob as *const $name as *const $crate::PyAny)}
+                unsafe { &*(ob as *const $name as *const $crate::PyAny) }
             }
         }
-    }
+
+        impl From<&'_ $name> for $crate::Py<$name> {
```

Comment:
I think it would be nice for `rust-numpy` to have this impl once it is possible, so maybe can we add a FIXME to move this back to `pyobject_native_type_named` once we bump MSRV to Rust 1.40 or higher?

Reply:
:+1:

Reply:
But why is this in `named`? I think it's more of a `convert` thing.

Reply:
:+1: